includes/datastream.inc

```diff
@@ -295,6 +295,12 @@ function islandora_datastream_get_url(AbstractDatastream $datastream, $type = 'd
  */
 function islandora_edit_datastream(AbstractDatastream $datastream) {
   $edit_registry = module_invoke_all('islandora_edit_datastream_registry', $datastream->parent, $datastream);
+  $context = array(
+    'datastream_parent' => $datastream->parent,
```

Comment:
@matthewperry, little note: maybe name this key in `$context` array  `object` instead of `datastream_parent `, since it's the object.

Also, in case  `ISLANDORA_EDIT_DATASTREAM_MODIFY_REGISTRY_HOOK` ends unsetting all routes, maybe it would be better to put some check after this lines also? So you don't show the link at all? https://github.com/Islandora/islandora/blob/7.x/theme/theme.inc#L537